WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -32,18 +33,30 @@ public LabelSelectionViewModel(Money targetAmount)
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] AutoSelectPockets()
+	public Pocket[] AutoSelectPockets(SmartLabel recipient)
 	{
 		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
 		var unknownPockets = NonPrivatePockets.Except(knownPockets).ToArray();
 		var privateAndUnknownPockets = _allPockets.Except(knownPockets).ToArray();
 		var privateAndKnownPockets = _allPockets.Except(unknownPockets).ToArray();
+		var knownByRecipientPockets = knownPockets.Where(pocket => pocket.Labels.Any(recipient.Contains)).ToArray();
+		var onlyKnownByRecipientPockets = knownByRecipientPockets.Where(pocket => pocket.Labels == recipient).ToArray();
+
+		if (onlyKnownByRecipientPockets.Sum(x => x.Amount) >= _targetAmount)
+		{
+			return onlyKnownByRecipientPockets;
+		}
 
 		if (_privatePocket.Amount >= _targetAmount)
 		{
 			return new[] { _privatePocket };
 		}
 
+		if (GetBestKnownByRecipientPockets(knownByRecipientPockets, _targetAmount, recipient) is { } pockets)
```

Comment:
We should check the sum of the amounts here:

```suggestion
		if (GetBestKnownByRecipientPockets(knownByRecipientPockets, _targetAmount, recipient) is { } pockets && pockets.Sum(x => x.Amount) >= _targetAmount)
```

Otherwise this test wouldn't pass:

```
public void MyTest()
{
	var selection = new LabelSelectionViewModel(Money.Parse("1.0"));

	var pockets = new List<Pocket>();
	pockets.AddPocket(0.8M, out var pocket1, CoinPocketHelper.PrivateFundsText);
	pockets.AddPocket(1.1M, out var pocket2, "Dan");
	pockets.AddPocket(0.5M, out var pocket3, CoinPocketHelper.UnlabelledFundsText);
	pockets.AddPocket(0.1M, out var pocket4, "David", "Lucas");
	pockets.AddPocket(0.15M, out var pocket5, "David");
	pockets.AddPocket(0.2M, out var pocket6, "Lucas");
	pockets.AddPocket(0.3M, out var pocket7, "David", "Lucas", "Dan");

	selection.Reset(pockets.ToArray());

	var output = selection.AutoSelectPockets(new SmartLabel("David", "Lucas"));
	Assert.DoesNotContain(pocket1, output);
	Assert.Contains(pocket2, output);
	Assert.DoesNotContain(pocket3, output);
	Assert.Contains(pocket4, output);
	Assert.Contains(pocket5, output);
	Assert.Contains(pocket6, output);
	Assert.Contains(pocket7, output);
}
```

If you agree this test is a better replacement to the `AutoSelectOnlyUnknownPocket` test.

Reply:
> We should check the sum of the amounts here:

Fixed in another way, I made a mistake when I checked if the search was successful or not.

> If you agree this test is a better replacement to the AutoSelectOnlyUnknownPocket test.

You mean `AutoSelectOnlyKnownPockets`?